admin/modules/config/plugins.php

```diff
@@ -517,36 +680,53 @@
 
 	$plugins->run_hooks("admin_config_plugins_plugin_list");
 
+	$form = new Form('index.php?module=config-plugins&amp;action=upload', 'post', '', 1);
+	$table = new Table;
+	$table->construct_header($lang->upload_plugin_desc);
+	$table->construct_cell($form->generate_file_upload_box('plugin_upload', array('style' => 'width: 230px;')).
+	                       $form->generate_check_box('ignore_vers', 1, $lang->plugin_ignore_vers).
+	                       ' '.
+	                       $form->generate_submit_button($lang->install_uploaded_plugin));
+	$table->construct_row();
+	$table->output($lang->upload_plugin);
+	$form->end();
+
 	if(!empty($plugins_list))
 	{
 		$a_plugins = $i_plugins = array();
 
-		foreach($plugins_list as $plugin_file)
+		if(!empty($plugins_list))
```

Comment:
unless github is messing up the diff view, this condition is superfluous as it's literally done 4 lines above.